ui/scripts/storage.js

```diff
@@ -1575,13 +1575,13 @@
                                     if (newDiskOffering != null && newDiskOffering.length > 0) {
                                         array1.push("&diskofferingid=" + encodeURIComponent(newDiskOffering));
                                     }
-                                    if (selectedDiskOfferingObj.iscustomized == true) {
+                                    if (selectedDiskOfferingObj == null || selectedDiskOfferingObj.iscustomized == true) {
```

Comment:
aren't we making an assumption here about the reason the offering is missing? We only want to allow it if it is missing because it is a root-disk. It looks like we are allowing some client side hacking; set the `selectedDiskOfferingObj` to `null` and you are good to go. (i didn't check the API)

Reply:
I have changed the logic to check the volume type instead.